src/routes/users.routes.js

```diff
@@ -1,11 +1,15 @@
 const router = require('express').Router();
-const { userController } = require('../controllers');
-const { validate } = require('../middlewares');
+const { userController,  sessionController } = require('../controllers');
+const { validate, ensureAuthenticated } = require('../middlewares');
+
 const {
   validationSchemas: { users }
 } = require('../validations');
 
-router.post('/create', validate(users.create) ,userController.create);
+router.post('/login', validate(users.session), sessionController.Authenticate);
+
+router.post('/create', validate(users.create),ensureAuthenticated.auth, userController.create);
```

Comment:
Se for ter várias rotas que precisam de autenticação, vale a pena colocar `router.use(ensureAuthenticated.auth);` na linha de cima

Reply:
aplicado